sos/upload/targets/__init__.py

```diff
@@ -540,8 +540,9 @@ def upload_sftp(self, user=None, password=None):
             raise Exception("Unable to connect via SFTP to "
                             f"{self.get_upload_url_string()}")
 
+        # certain implementations require file to be put in the user dir
         put_cmd = (f'put {self.upload_archive_name} '
-                   f'{self._get_sftp_upload_name()}')
+                   f'{user_dir}/{self._get_sftp_upload_name()}')
```

Comment:
The trailing `/` might throw off instances where no `user_dir` is set. E.G. `put sos.tar mysos.tar` would become `put sos.tar /mysos.tar`.

What about:
```python
put_cmd = f"put {self.upload_archive_name} {f'{user_dir}/' if user_dir else ''}{self._get_sftp_upload_name()}"
```

Then in the Red Hat class we'd have:

```python
return super().upload_sftp(user=_user, password=_token, user_dir=_user)
```